src/OpenSpaceToolkit/Physics/Environment/Ephemeris/SPICE/Engine.cpp

```diff
@@ -351,20 +351,11 @@ void Engine::setup()
 
     errdev_c("SET", 4096, (SpiceChar*)"NULL");
 
-    if (mode_ == Engine::Mode::Automatic)
-    {
-        Directory kernelDirectory = Manager::Get().getLocalRepository();
-        if (!kernelDirectory.exists())
-        {
-            kernelDirectory.create();
-        }
-
-        // Load default kernels
+    // Load default kernels
```

Comment:
You don't need to lock a mutex here? I see it being used above before calling the `this->loadKernel_(kernel);` function

Reply:
The setup function is called in the constructor at which point no mutex is required I believe as there is only a single instance of the Engine. And the second way the setup function is called is in reset which has it's own lock guard.